token/js/src/extensions/transferHook/state.ts

```diff
@@ -49,3 +52,68 @@ export function getTransferHookAccount(account: Account): TransferHookAccount |
         return null;
     }
 }
+
+/** ExtraAccountMeta as stored by the transfer hook program */
+export interface ExtraAccountMeta {
+    discriminator: number;
+    addressConfig: Uint8Array;
+    isSigner: boolean;
+    isWritable: boolean;
+}
+
+/** Buffer layout for de/serializing an ExtraAccountMeta */
+export const ExtraAccountMetaLayout = struct<ExtraAccountMeta>([
+    u8('discriminator'),
+    blob(32, 'addressConfig'),
+    bool('isSigner'),
+    bool('isWritable'),
+]);
+
+export interface ExtraAccountMetaList {
+    count: number;
+    extraAccounts: ExtraAccountMeta[];
+}
+
+/** Buffer layout for de/serializing a list of ExtraAccountMeta prefixed by a u32 length */
+export const ExtraAccountMetaListLayout = struct<ExtraAccountMetaList>([
+    u32('count'),
+    seq<ExtraAccountMeta>(ExtraAccountMetaLayout, greedy(ExtraAccountMetaLayout.span), 'extraAccounts'),
+]);
+
+/** Unpack an extra account metas account and parse the data into a list of ExtraAccountMetas */
+export function getExtraAccountMetas(account: AccountInfo<Buffer>): ExtraAccountMeta[] | null {
+    return ExtraAccountMetaListLayout.decode(account.data).extraAccounts;
+}
```

Comment:
This won't quite work unfortunately -- it's taking in the full account data buffer and interpreting those bytes as the extra account metas.

The extra account metas are actually encoded as a TLV structure, where `type` is 8 bytes, `length` is 4 bytes, and the `value` is the following `length` bytes.

So in your case, for the `TransferHookInstruction::Execute` instruction, you're looking for the discriminant `[105, 37, 101, 197, 75, 251, 102, 26]`. You'll take the first 8 bytes as the type, then the next 4 bytes as the length. If the first 8 bytes match, then you decode the next `length` bytes (after the length) as `ExtraAccountMetaList`. If they don't match, you'll skip those next `length` bytes, then read the next 8 bytes as the type, 4 bytes as the length, etc etc

Also, be sure to *not* return the whole slice as `ExtraAccountMetas`, just the first `count` items, ie:

```
const decoded = ExtraAccountMetaListLayout.decode(account.data);
return decoded.extraAccounts.slice(0, decoded.count);
```

Reply:
I don't like merging code that we know doesn't work -- can you at least port the bit to skip the first 12 bytes?